receiver/windowsservicereceiver/windowsservice.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+//go:build windows
+
+package windowsservicereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowsservicereceiver"
+
+import (
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+// represents the service struct before we have converted it into a metric
+type ServiceStatus struct {
+	ServiceStatus uint32
+	StartType     uint32
+	service       *mgr.Service
+}
+
+func GetService(sname string) (*ServiceStatus, error) {
+	m, err := SCConnect()
+	defer m.Disconnect()
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := m.OpenService(sname)
+	if err != nil {
+		return nil, err
+	}
+
+	s := ServiceStatus{
+		service: service,
+	}
+
+	// populate metric fields
+	if err = s.getStatus(); err != nil {
+		return nil, err
+	}
+
+	if err = s.getConfig(); err != nil {
+		return nil, err
+	}
+
+	return &s, nil
+}
+
+// populates fields from service status query
+func (s *ServiceStatus) getStatus() error {
+	st, err := s.service.Query()
+	if err != nil {
+		return err
```

Comment:
This is not a problem with the current code since the handle to the service is opened every time a `GetService` call is made. However, it is good to pay attention here that the value of `s.ServiceStatus` in case of error will be the default value for the type (0 in this case).

Reply:
Similar in the call to `Config` below.